src/components/common/needsShutdown.jsx

```diff
@@ -0,0 +1,161 @@
+            (vm.cpu.cores !== vm.inactiveXML.cpu.cores));
+}
+
+export function needsShutdownCpuModel(vm) {
+    /* The live xml shows what host-model expanded to when started
+     * This is important since the expansion varies depending on the host and so needs to be tracked across migration
+     */
+    if (vm.inactiveXML.cpu.mode === 'host-model')
+        return !(vm.cpu.mode == 'host-model' || vm.cpu.model === vm.capabilities.cpuHostModel);
+    if (vm.inactiveXML.cpu.mode === 'host-passthrough')
+        return vm.cpu.mode !== 'host-passthrough';
+    if (vm.inactiveXML.cpu.mode === 'custom')
+        return vm.cpu.mode !== 'custom' || vm.cpu.model !== vm.inactiveXML.cpu.model;
+
+    return false;
+}
+
+export function needsShutdownWatchdog(vm) {
+    return vm.persistent && vm.state === "running" && vm.inactiveXML.watchdog.action !== vm.watchdog.action;
+}
+
+export function getDevicesRequiringShutdown(vm) {
+    if (!vm.persistent)
+        return [];
+
+    const devices = [];
+
+    // DISKS
+    for (const target in vm.disks) {
+        if (needsShutdownDiskAccess(vm, target)) {
+            devices.push(_("Disk"));
+            break;
+        }
+    }
+
+    // INTERFACES
+    for (const iface of vm.interfaces) {
+        if (needsShutdownIfaceType(vm, iface) ||
+            needsShutdownIfaceModel(vm, iface) ||
+            needsShutdownIfaceSource(vm, iface)) {
+            devices.push(_("Network interface"));
+            break;
+        }
+    }
+
+    // VCPU
+    if (needsShutdownVcpu(vm))
+        devices.push(_("vCPUs"));
+
+    // CPU
+    if (needsShutdownCpuModel(vm))
+        devices.push(_("CPU"));
+
+    // Watchdog
+    if (needsShutdownWatchdog(vm))
+        devices.push(_("Watchdog"));
+
+    return devices;
+}
+
+export const NeedsShutdownTooltip = ({ iconId, tooltipId }) => {
+    return (
+        <Tooltip id={tooltipId} content={NEEDS_SHUTDOWN_MESSAGE}>
+            <PendingIcon color="orange" id={iconId} />
+        </Tooltip>
+    );
+};
+
+export const NeedsShutdownAlert = ({ idPrefix }) =>
+    <Alert isInline variant='warning' id={`${idPrefix}-idle-message`} title={NEEDS_SHUTDOWN_MESSAGE} />;
+
+export const VmNeedsShutdown = ({ vm }) => {
+    const devices = getDevicesRequiringShutdown(vm);
+
+    if (devices.length === 0)
+        return;
+
+    const body = (
+        <>
+            <List isPlain isBordered>
+                { devices.map(device => {
+                    return (
+                        <ListItem key={device}>
+                            {cockpit.format(_("$0 configuration was changed"), device)}
+                        </ListItem>
+                    );
+                }) }
+            </List>
+        </>
+    );
+
+    const header = _("VM needs shutdown");
+    return (
+        <Popover aria-label={header}
+            alertSeverityVariant="warning"
+            headerContent={header}
+            headerIcon={<ExclamationTriangleIcon />}
+            headerComponent="h1"
```

Comment:
<h1> headings should be used for main heading of a page. This is definitely not a main heading. Why did you want to change whatever it uses by default?

Reply:
Copied from patternfly example, didn't stop to think about it :D 
Fixed

Reply:
Yep. You're not supposed to change the order of headings.

Reply:
Patternfly's Popover uses by default --pf-c-popover__title--FontSize which is 1rem by default, and looks like this:
![Screenshot from 2023-04-03 12-17-40](https://user-images.githubusercontent.com/42733240/229483802-d2781ab4-7229-4fda-9b49-ae5007809e71.png)

So I think we can just keep patterfly's default